updates adb helper to cope with no space after word device

moshea opened the pull request:
adb seems to have changed its output from
"emulator-5554     device " to
"emulator-5554     device"
note the space after the work device

change uses a regular expression to match the word device with or
without a space after it, but does not match the word devices

Commit:
updates adb helper to cope with no space after word device

adb seems to have changed its output from
"emulator-5554     device " to
"emulator-5554     device"
note the space after the work device

change uses a regular expression to match the word device with or
without a space after it, but does not match the word devices
also takes into account that there is a space after the word device
for a physical device, such as
"192.168.56.101:5555 device product:vbox86p model:device1 device:vbox86p"

rajdeepv commented:
the parsing is happening on `adb devices -l` instead of `adb devices` 

`adb devices`  gives output without space but `adb devices -l` gives output with spaces as shown:

rajdeep:~ rajdeepverma$ adb devices -l
List of devices attached
192.168.56.101:5555    device product:vbox86p model:device1 device:vbox86p
192.168.56.102:5555    device product:vbox86p model:device2 device:vbox86p

nothing is matching when I do this:
 "List of devices attached \n192.168.56.101:5555    device product:vbox86p model:device1 device:vbox86p\n192.168.56.102:5555    device product:vbox86p model:device2 device:vbox86p\n\n".match(/device ?$/)

moshea commented:
Ah yes, I see the issue. I've updated the regular expression to take the physical devices into account.
If you are only using android emulators, `adb devices -l` does not leave a trailing space after the word device. you will get "emulator-5554 device"
example:

ngmosxmm2:~ jenkins$ adb devices -l
List of devices attached 
emulator-5554          device

i've also added a test which takes the physical device into account

rajdeepv commented:
Thanks @moshea

moshea commented:
Thanks Rajdeep!

On Thu, Apr 30, 2015 at 5:10 PM, Rajdeep notifications@github.com wrote:

> Thanks @moshea https://github.com/moshea
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/rajdeepv/parallel_calabash/pull/6#issuecomment-97859940
> .

rajdeepv commented:
@moshea can you please test this in v0.1.2?
